### DISCUSSION TITLE
cyber logon with non interactive password fails

### DISCUSSION THREAD

**Discussion by timoheld** [infamousjoeg/cybr-cli#226] [General]
When I try to log in with a password, it always fails with the message "Provided password is empty". 
What am I doing wrong?
```
cybr logon -u <user> -a cyberark --non-interactive --password "<password>"  -b <url>
2024/05/08 15:54:08 Provided password is empty

cybr version
cybr v1.0.2-release
```

**Comment by ghost** [comment]
Per https://github.com/infamousjoeg/cybr-cli/blob/main/docs/cybr_logon.md:

Non-interactive mode description states the password is referenced as an environment variable.  I believe this implies the --password argument is now deprecated.

The risk of passing the password via command line argument is may lead to your shell interpreter capturing/logging the password for the account.

**Reply by timoheld** [reply]
Thanks for your reply. 

I tried using the environment variable, but i got the same error: 

`PAS_PASSWORD="<password>"
cybr logon -u <user> -a cyberark --non-interactive  -b <url>
 2024/05/14 10:17:42 Provided password is empty
`

then i tried using with the non interactive flag and the password flag, but i got the same error

**Reply by timoheld** [reply]
hi everyone, does anyone know why the top mentioned, is not working?

`PAS_PASSWORD="<password>" 
cybr logon -u <user> -a cyberark --non-interactive  -b <url> 
2024/05/14 10:17:42 Provided password is empty`

i also tried to add -p 

`PAS_PASSWORD="<password>" 
cybr logon -u <user> -a cyberark --non-interactive  -b <url> -p <password>
2024/05/14 10:17:42 Provided password is empty`
